hazelcast/src/main/java/com/hazelcast/client/impl/protocol/ClientMessage.java

```diff
@@ -328,7 +328,9 @@ public boolean readFrom(ByteBuffer src) {
                 // we don't have even the frame length ready
                 return false;
             }
-            frameLength = Bits.readIntL(src.array(), src.position());
+            frameLength = Bits.readIntL(src);
```

Comment:
It's the only place you read from the provided byteBuffer so you don't need to store and restore any endianness state. But if you don't like it we can keep it, it's a minor thing